userbenchmark/cpu/__init__.py

```diff
@@ -0,0 +1,144 @@
+        device=device,
+        test=test,
+        batch_size=batch_size,
+        jit=jit,
+        extra_args=extra_args,
+        extra_env=None,
+    ) for device, test, model_name in cfgs]
+    return result
+
+def dump_result_to_json(metrics, output_dir, fname):
+    result = get_output_json(BM_NAME, metrics)
+    dump_output(BM_NAME, result, output_dir, fname)
+
+def validate(candidates: List[str], choices: List[str]) -> List[str]:
+    """Validate the candidates provided by the user is valid"""
+    for candidate in candidates:
+        assert candidate in choices, f"Specified {candidate}, but not in available list: {choices}."
+    return candidates
+
+def generate_model_configs_from_yaml(yaml_file: str) -> List[TorchBenchModelConfig]:
+    yaml_file_path = os.path.join(CURRENT_DIR, yaml_file)
+    with open(yaml_file_path, "r") as yf:
+        config_obj = yaml.safe_load(yf)
+    devices = config_obj.keys()
+    configs = []
+    for device in devices:
+        for c in config_obj[device]:
+            if not c["stable"]:
+                continue
+            config = TorchBenchModelConfig(
+                name=c["model"],
+                device=device,
+                test=c["test"],
+                batch_size=c["batch_size"] if "batch_size" in c else None,
+                jit=c["jit"] if "jit" in c else False,
+                extra_args=[],
+                extra_env=None,
+            )
+            configs.append(config)
+    return configs
+
+def parse_str_to_list(candidates):
+    if isinstance(candidates, list):
+        return candidates
+    candidates = list(map(lambda x: x.strip(), candidates.split(",")))
+    return candidates
+
+def parse_args(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--device", "-d", default="cpu", help="Devices to run, splited by comma.")
+    parser.add_argument("--test", "-t", default="eval", help="Tests to run, splited by comma.")
+    parser.add_argument("--model", "-m", default=None, help="Only run the specifice models, splited by comma.")
+    parser.add_argument("--batch-size", "-b", default=None, help="Run the specifice batch size.")
+    parser.add_argument("--jit", action="store_true", help="Convert the models to jit mode.")
+    parser.add_argument("--config", "-c", default=None, help="YAML config to specify tests to run.")
+    parser.add_argument("--output", "-o", default=None, help="Output dir.")
+    parser.add_argument("--launcher", action="store_true", help="Use torch.backends.xeon.run_cpu to get the peak performance on Intel(R) Xeon(R) Scalable Processors.")
+    parser.add_argument("--launcher-args", default=None, help="Provide the args of torch.backends.xeon.run_cpu. See `python -m torch.backends.xeon.run_cpu --help`")
+    parser.add_argument("--dryrun", action="store_true", help="Dryrun the command.")
+    return parser.parse_known_args()
+
+def run(args: List[str]):
+    args, extra_args = parse_args(args)
+    extra_args.remove(BM_NAME)
```

Comment:
This can be removed, if we only parse `args`.

Reply:
Correct, thanks to point it out, will update it